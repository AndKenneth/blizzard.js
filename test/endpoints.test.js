/* global describe, context, it */
'use strict';

const path = require('path');
const chai = require('chai');
const endpoints = require(path.normalize(`${__dirname}/../lib/endpoints`));

describe('lib/endpoints.js', function () {

  it('should have a method `getEndpoint`', function (done) {
    chai.assert.isFunction(endpoints.getEndpoint);
    done();
  });

  context('.getEndpoint()', function () {
    const endpoint = endpoints.getEndpoint();

    it('should return the default endpoint', function (done) {
      chai.assert.deepEqual(endpoint, {
        hostname: 'https://us.api.battle.net',
        defaultLocale: 'en_US',
        locales: ['en_US', 'es_MX', 'pt_BR']
      });
      done();
    });
  });

  context('.getEndpoint(`eu`)', function () {
    const endpoint = endpoints.getEndpoint('eu');

    it('should return the requested endpoint', function (done) {
      chai.assert.deepEqual(endpoint, {
        hostname: 'https://eu.api.battle.net',
        defaultLocale: 'en_GB',
        locales: ['en_GB', 'es_ES', 'fr_FR', 'ru_RU', 'de_DE', 'pt_PT', 'it_IT']
      });
      done();
    });
  });

});
