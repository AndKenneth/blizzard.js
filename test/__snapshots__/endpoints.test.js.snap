// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/endpoints.js #getEndpoint() with invalid parameters should return the default endpoint 1`] = `
Object {
  "hostname": "https://us.api.battle.net",
  "locale": "en_US",
}
`;

exports[`lib/endpoints.js #getEndpoint() with no parameters should return the default endpoint 1`] = `
Object {
  "hostname": "https://us.api.battle.net",
  "locale": "en_US",
}
`;

exports[`lib/endpoints.js #getEndpoint() with valid parameters should return the EU es_ES endpoint 1`] = `
Object {
  "hostname": "https://eu.api.battle.net",
  "locale": "es_ES",
}
`;

exports[`lib/endpoints.js #getEndpoint() with valid parameters should return the SEA endpoint 1`] = `
Object {
  "hostname": "https://sea.api.battle.net",
  "locale": "en_US",
}
`;
